### ISSUE TITLE
Vendored flag behaves differently than when using system libssh when connecting to bitbucket

### ISSUE THREAD

**Issue by JetpackDuba** [wez/libssh-rs#18]
Hey!

Before anything else, thank you for your work in this crate :smile: 

I'm using it in another FOSS project to use SSH connections and it worked nicely so far until this issue was [reported](https://github.com/JetpackDuba/Gitnuro/issues/166). When trying to establish a connection with Bitbucket it fails complaining with a "Socket error: disconnected".

I've spent a few days trying to find what was the error as the code should be the same as what I had written in C before but nothing worked (the C version worked nicely). Then I tried a new project to try to make a reproducible sample but surprisingly it worked, being the only difference between the original project and this new one the fact that the "vendored" feature is active on the first one but not on the second.

In other words, when the _vendored_ flag is active, this code fails:
```
use libssh_rs::SshOption;

const ACCEPTED_SSH_TYPES: &str = "ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,rsa-sha2-512,rsa-sha2-256,ssh-dss";

fn main() {
    let session = libssh_rs::Session::new().unwrap();

    session.set_option(SshOption::Hostname("bitbucket.org".to_string())).unwrap();
    session.set_option(SshOption::PublicKeyAcceptedTypes(ACCEPTED_SSH_TYPES.to_string())).unwrap();
    session.connect().unwrap();

    let status = session.userauth_public_key_auto(Some("git"), None).unwrap();
    println!("Status is {status:#?}");

    let channel = session.new_channel().unwrap();
    channel.open_session().unwrap();
    channel.request_exec("dummy command").unwrap(); // fails here but sometimes even in the previous line, not sure why.
}
```

This cargo config fails:
```
[package]
name = "tests_rs"
version = "0.1.0"
edition = "2021"

[dependencies]
libssh-rs = { version = "0.2.1", features = [ "vendored", "vendored-openssl" ] }
```

This one works:
```
[package]
name = "tests_rs"
version = "0.1.0"
edition = "2021"

[dependencies]
libssh-rs = { version = "0.2.1", features = [ "vendored-openssl" ] }
```

Removing the feature to use the system version of libssh works fine (The installed version on my system is 0.10.5, which should be the same as in libssh-rs-sys). This can be kinda ok on Linux but on Windows and MacOS makes it quite a pain to build (I've test on Windows and the exact same issue shows up).

To test it you need to add your SSH key to your bitbucket account, otherwise it will fail before reaching the problematic line.

Not sure what can be wrong but let me know if I can help somehow.

Thanks,
Abde.

**Comment by JetpackDuba** [comment]
Hey @wez ,

After spending a good bunch of hours in this issue, I think I've found the issue. When using a statically built version of libssh with CMake it worked just fine, after a lot of digging and comparing, I've found that this issue shows up when building libssh with the flag `WITH_SERVER` disabled/off (Enabled by default when building with CMake). After enabling it and adding the server.c file to the list of files, it worked perfectly (tested on ArchLinux and Windows). 

Not sure if using the [CMake](https://crates.io/crates/cmake) instead of the CC crate could be an option, since it's the build system used by LibSSH.
